src/Entity/ProductBundle.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\commerce_product\Entity\ProductBundle;
+
+use Drupal\commerce\PurchasableEntityInterface;
+use Drupal\commerce_product\Entity\Product;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Defines the product entity class.
+ *
+ * @ContentEntityType(
+ *   id = "commerce_product_bundle",
+ *   label = @Translation("Product Bundle"),
+ *   label_singular = @Translation("product bundle"),
+ *   label_plural = @Translation("product bundles"),
+ *   label_count = @PluralTranslation(
+ *     singular = "@count product bundle",
+ *     plural = "@count product bundles",
+ *   ),
+ *   handlers = {
+ *     "event" = "Drupal\commerce_product\Event\ProductEvent",
```

Comment:
we'll have our own events, or at least commerce_product_bundle\Event\BundleEvent